src/content/marina/marinaBroker.ts

```diff
@@ -394,48 +420,31 @@ export default class MarinaBroker extends Broker<keyof Marina> {
             throw new Error('Only custom script accounts can import templates');
           }
 
-          const [covenant, changeCovenant] = params?.map(castTemplate) as [Template, Template?];
-          if (!validate(covenant.template)) {
-            throw new Error(`Invalid template ${covenant.template}`);
-          }
-
-          if (changeCovenant && !validate(changeCovenant.template)) {
-            throw new Error(`Invalid change template ${changeCovenant.template}`);
+          let contract: Template | undefined;
+          if (params && params.length > 0) {
+            if (!validateTemplate(params[0])) {
+              throw new Error('Invalid template');
+            }
+            contract = params[0];
           }
 
           await this.store.dispatchAsync(
-            setCustomScriptTemplate(
-              this.selectedAccount,
-              covenant.template,
-              changeCovenant?.template
-            )
+            setCustomScriptTemplate(this.selectedAccount, contract!.template)
           );
 
-          const selectedAccount = selectAccount(this.selectedAccount)(this.state);
-          const watchIdentity = await selectedAccount.getWatchIdentity(selectNetwork(this.state));
-
-          const nextAddress = (await watchIdentity.getNextAddress()) as TaprootAddressInterface;
-          const autoSpendableLeaf = (needsOfLeaf: ScriptInputsNeeds) =>
-            needsOfLeaf.sigs.length === 1 &&
-            !needsOfLeaf.needParameters &&
-            !needsOfLeaf.hasIntrospection;
-          let isSpendableByMarina = Object.values(
-            analyzeTapscriptTree(nextAddress.taprootHashTree)
-          ).some(autoSpendableLeaf);
-
-          if (changeCovenant) {
-            const nextChangeAddress =
-              (await watchIdentity.getNextChangeAddress()) as TaprootAddressInterface;
-            isSpendableByMarina ||= Object.values(
-              analyzeTapscriptTree(nextChangeAddress.TaprootAddressInterface)
-            ).some(autoSpendableLeaf);
-          }
+          const artifact: Artifact = JSON.parse(contract!.template);
+          const isSpendableByMarina = artifact.functions.every(({ functionInputs, require }) => {
+            return (
+              functionInputs.length === 1 &&
+              functionInputs[0].type === PrimitiveType.Signature &&
+              (!require || require.length === 0)
```

Comment:
nit: require could lead to conflict with the JS require :thinking:

Reply:
@altafan